web/src/routes/(user)/utilities/duplicates/[[photos=photos]]/[[assetId=id]]/+page.ts

```diff
@@ -1,13 +1,33 @@
 import { authenticate } from '$lib/utils/auth';
 import { getFormatter } from '$lib/utils/i18n';
 import { getAssetInfoFromParam } from '$lib/utils/navigation';
-import { getAssetDuplicates } from '@immich/sdk';
+import { getAssetDuplicates, updateAssets, type DuplicateResponseDto } from '@immich/sdk';
 import type { PageLoad } from './$types';
 
+const rectifyDuplicate = (uniqueAssetIds: string[]) => {
+  if (uniqueAssetIds.length > 0) {
+    void updateAssets({ assetBulkUpdateDto: { ids: uniqueAssetIds, duplicateId: null } });
+  }
+};
+
+const processDuplicates = (duplicates: DuplicateResponseDto[]) => {
+  const uniqueAssetIds: string[] = [];
+  const validDuplicates = duplicates.filter((duplicate) => {
+    if (duplicate.assets.length <= 1) {
+      uniqueAssetIds.push(duplicate.assets[0].id);
+      return false;
+    }
+    return true;
+  });
+  rectifyDuplicate(uniqueAssetIds);
+  return validDuplicates;
+};
+
 export const load = (async ({ params }) => {
   await authenticate();
   const asset = await getAssetInfoFromParam(params);
-  const duplicates = await getAssetDuplicates();
+  const Allduplicates = await getAssetDuplicates();
+  const duplicates = processDuplicates(Allduplicates);
```

Comment:
This is up to preference, but I think it'd be cleaner if this returned the valid and unique duplicates and you call `rectifyDuplicates` on the latter.

Reply:
Do we mean the getAssetDuplicates function to return these two lists?

Reply:
Yup!